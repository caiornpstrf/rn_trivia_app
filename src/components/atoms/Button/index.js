/**
 * @summary Custom button with multiple styles to chose from.
 * @author Caio Reis <caio.oliveira.reis@gmail.com>
 *
 * Created at     : 2021-04-11 19:57:42
 * Last modified  : 2021-04-16 04:19:34
 */
import React from 'react';
import {Text, Pressable} from 'react-native';
import {func, oneOf, string} from 'prop-types';

import selectStyles from './selectStyles';

const Button = ({label, type, onPress, style, ...rest}) => {
  const styles = selectStyles(type);
  return (
    <Pressable
      style={({pressed}) => [
        styles.container,
        style,
        pressed && {opacity: 0.8},
      ]}
      onPress={onPress}
      {...rest}>
      <Text style={styles.text}>{label}</Text>
    </Pressable>
  );
};

Button.propTypes = {
  label: string,
  type: oneOf(['default', 'answer']),
  onPress: func,
};

Button.defaultProps = {
  label: '',
  type: 'default',
  onPress: () => console.log('Pressed Button'),
};

export default Button;
