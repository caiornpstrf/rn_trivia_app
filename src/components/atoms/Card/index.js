/**
 * @summary Generic Card with built-in shadow
 * @author Caio Reis <caio.oliveira.reis@gmail.com>
 *
 * Created at     : 2021-04-16 03:34:10
 * Last modified  : 2021-04-16 03:34:10
 */

import React from 'react';
import {View} from 'react-native';

import styles from './styles';

const Card = ({children, style}) => {
  return <View style={[styles.container, style]}>{children}</View>;
};

export default Card;
