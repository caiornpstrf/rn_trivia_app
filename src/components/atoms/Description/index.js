/**
 * @summary Standardized Text component to fit descriptions
 * @author Caio Reis <caio.oliveira.reis@gmail.com>
 *
 * Created at     : 2021-04-16 03:34:22
 * Last modified  : 2021-04-16 03:34:50
 */

import React from 'react';
import {StyleSheet, Text} from 'react-native';

import {Colors, StyleHelpers} from '_assets/styles';

const Description = ({children}) => {
  return <Text style={styles.text}>{children}</Text>;
};

export default Description;

const styles = StyleSheet.create({
  text: {
    color: Colors.text,
    fontSize: StyleHelpers.normalize(12),
  },
});
