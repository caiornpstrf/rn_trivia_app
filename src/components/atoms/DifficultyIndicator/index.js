/**
 * @summary Indicates difficulty using stars.
 * @author Caio Reis <caio.oliveira.reis@gmail.com>
 *
 * Created at     : 2021-04-11 16:20:17
 * Last modified  : 2021-04-12 03:51:35
 */

import React from 'react';
import {StyleSheet, Text, View} from 'react-native';
import {string} from 'prop-types';

import {Icon} from '_components/atoms';
import {Colors} from 'assets/styles';

const difficultyMapping = {
  easy: 0,
  medium: 1,
  hard: 2,
};

const renderIconList = (currentDifficultyNumber, maxDifficulty) => {
  let iconsList = [];
  for (let i = 0; i < maxDifficulty; i++) {
    iconsList.push(
      <Icon
        key={'icon_' + i}
        name="star"
        color={i <= currentDifficultyNumber ? Colors.title : Colors.disabled}
      />,
    );
  }

  return iconsList;
};

const DifficultyIndicator = ({currentDifficulty /* , maxDifficulty */}) => {
  const iconsList = renderIconList(difficultyMapping[currentDifficulty], 3);
  return (
    <View style={styles.container}>
      <View style={styles.stars}>{iconsList}</View>
      <Text style={styles.text}>{currentDifficulty}</Text>
    </View>
  );
};

DifficultyIndicator.propTypes = {
  currentDifficulty: string,
  //maxDifficulty: number,
};

DifficultyIndicator.defaultProps = {
  currentDifficulty: 'medium',
  //maxDifficulty: 3,
};

export default DifficultyIndicator;

const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    maxHeight: 30,
    minWidth: 60,
    borderRadius: 20,
    paddingVertical: 5,
    paddingHorizontal: 10,
    backgroundColor: Colors.altBackground,
  },
  stars: {
    flexDirection: 'row',
  },
  text: {
    color: Colors.title,
    textTransform: 'capitalize',
  },
});
