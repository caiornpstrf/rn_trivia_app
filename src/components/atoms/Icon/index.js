/**
 * @summary Custom Icon made using Vector Icons as base
 * @author Caio Reis <caio.oliveira.reis@gmail.com>
 *
 * Created at     : 2021-04-12 02:59:55
 * Last modified  : 2021-04-16 03:35:41
 */

import React from 'react';
import {number, string} from 'prop-types';

import Icon from 'react-native-vector-icons/FontAwesome';

const CustomIcon = ({name, size, color, ...rest}) => {
  return <Icon name={name} size={size} color={color} {...rest} />;
};

CustomIcon.propTypes = {
  name: string.isRequired,
  size: number,
  color: string.isRequired,
};

CustomIcon.defaultProps = {
  size: 12,
};

export default CustomIcon;
