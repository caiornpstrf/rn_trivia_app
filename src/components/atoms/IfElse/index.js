/**
 * @summary Simple loop component to render multiple children
 * @author Caio Reis <caio.oliveira.reis@gmail.com>
 *
 * Created at     : 2021-04-12 03:10:37
 * Last modified  : 2021-04-12 03:37:08
 */
import {bool, node} from 'prop-types';
import React from 'react';

const IfElse = ({condition, onTrue, onFalse}) => {
  if (condition) {
    return <>{onTrue}</>;
  }
  return <>{onFalse}</>;
};

IfElse.propTypes = {
  condition: bool.isRequired,
  onTrue: node.isRequired,
  onFalse: node.isRequired,
};

export default IfElse;
