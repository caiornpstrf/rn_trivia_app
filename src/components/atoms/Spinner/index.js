import React from 'react';
import {StyleSheet, ActivityIndicator, View} from 'react-native';

import {Colors} from '_assets/styles';

const Spinner = () => {
  return (
    <View style={styles.container}>
      <ActivityIndicator color={Colors.primary} />
    </View>
  );
};

export default Spinner;

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
  },
});
