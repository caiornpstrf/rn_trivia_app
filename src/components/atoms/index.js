export {default as Button} from './Button';
export {default as Card} from './Card';
export {default as Description} from './Description';
export {default as DifficultyIndicator} from './DifficultyIndicator';
export {default as Icon} from './Icon';
export {default as IfElse} from './IfElse';
export {default as Title} from './Title';
