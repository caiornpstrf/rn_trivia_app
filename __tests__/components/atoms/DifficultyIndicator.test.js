import React from 'react';
import {render} from '@testing-library/react-native';
import {DifficultyIndicator} from '_components/atoms';

const difficulties = {
  easy: 'easy',
  medium: 'medium',
  hard: 'hard',
};

it('renders on easy difficulty correctly', () => {
  const {toJSON} = render(
    <DifficultyIndicator currentDifficulty={difficulties.easy} />,
  );
  expect(toJSON()).toMatchSnapshot();
});

it('renders on medium difficulty correctly', () => {
  const {toJSON} = render(
    <DifficultyIndicator currentDifficulty={difficulties.medium} />,
  );
  expect(toJSON()).toMatchSnapshot();
});

it('renders on hard difficulty correctly', () => {
  const {toJSON} = render(
    <DifficultyIndicator currentDifficulty={difficulties.hard} />,
  );
  expect(toJSON()).toMatchSnapshot();
});
