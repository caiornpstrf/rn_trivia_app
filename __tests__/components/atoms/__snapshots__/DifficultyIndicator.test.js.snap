// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders on easy difficulty correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#D6D8DE",
      "borderRadius": 20,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "paddingHorizontal": 10,
      "paddingVertical": 5,
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
        "marginRight": 10,
      }
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "#343D58",
            "fontSize": 12,
          },
          undefined,
          Object {
            "fontFamily": "FontAwesome",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "#B0B0B0",
            "fontSize": 12,
          },
          undefined,
          Object {
            "fontFamily": "FontAwesome",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "#B0B0B0",
            "fontSize": 12,
          },
          undefined,
          Object {
            "fontFamily": "FontAwesome",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
  </View>
  <Text
    style={
      Object {
        "color": "#343D58",
        "textTransform": "capitalize",
      }
    }
  >
    easy
  </Text>
</View>
`;

exports[`renders on hard difficulty correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#D6D8DE",
      "borderRadius": 20,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "paddingHorizontal": 10,
      "paddingVertical": 5,
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
        "marginRight": 10,
      }
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "#343D58",
            "fontSize": 12,
          },
          undefined,
          Object {
            "fontFamily": "FontAwesome",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "#343D58",
            "fontSize": 12,
          },
          undefined,
          Object {
            "fontFamily": "FontAwesome",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "#343D58",
            "fontSize": 12,
          },
          undefined,
          Object {
            "fontFamily": "FontAwesome",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
  </View>
  <Text
    style={
      Object {
        "color": "#343D58",
        "textTransform": "capitalize",
      }
    }
  >
    hard
  </Text>
</View>
`;

exports[`renders on medium difficulty correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#D6D8DE",
      "borderRadius": 20,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "paddingHorizontal": 10,
      "paddingVertical": 5,
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
        "marginRight": 10,
      }
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "#343D58",
            "fontSize": 12,
          },
          undefined,
          Object {
            "fontFamily": "FontAwesome",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "#343D58",
            "fontSize": 12,
          },
          undefined,
          Object {
            "fontFamily": "FontAwesome",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "#B0B0B0",
            "fontSize": 12,
          },
          undefined,
          Object {
            "fontFamily": "FontAwesome",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
  </View>
  <Text
    style={
      Object {
        "color": "#343D58",
        "textTransform": "capitalize",
      }
    }
  >
    medium
  </Text>
</View>
`;
